contracts/protocol/modules/v1/AaveV3LeverageModule.sol

```diff
@@ -0,0 +1,1147 @@
+/*
+    Copyright 2022 Index Coop
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ReentrancyGuard } from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import { AaveV3 } from "../../integration/lib/AaveV3.sol";
+import { IAToken } from "../../../interfaces/external/aave-v2/IAToken.sol";
+import { IController } from "../../../interfaces/IController.sol";
+import { IDebtIssuanceModule } from "../../../interfaces/IDebtIssuanceModule.sol";
+import { IExchangeAdapter } from "../../../interfaces/IExchangeAdapter.sol";
+import { IPool } from "../../../interfaces/external/aave-v3/IPool.sol";
+import { IPoolAddressesProvider } from "../../../interfaces/external/aave-v3/IPoolAddressesProvider.sol";
+import { IModuleIssuanceHook } from "../../../interfaces/IModuleIssuanceHook.sol";
+import { IAaveProtocolDataProvider } from "../../../interfaces/external/aave-v3/IAaveProtocolDataProvider.sol";
+import { ISetToken } from "../../../interfaces/ISetToken.sol";
+import { IVariableDebtToken } from "../../../interfaces/external/aave-v2/IVariableDebtToken.sol";
+import { ModuleBase } from "../../lib/ModuleBase.sol";
+
+/**
+ * @title AaveV3LeverageModule
+ * @author Set Protocol / Index Coop
```

Comment:
did Set assist you on implementation?

Reply:
I did not interact with Set in writing this code, but the code is 90% copy paste of the original `AaveLeverageModule` which has `Set Protocol` as it's author.